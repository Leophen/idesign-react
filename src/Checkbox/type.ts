export interface CheckboxProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 按钮多选框尺寸
   * @default medium
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * 是否固定选中（受控）
   */
  checked?: boolean;
  /**
   * 是否默认选中（非受控）
   * @default false
   */
  defaultChecked?: boolean;
  /**
   * 是否禁用组件
   * @default false
   */
  disabled?: boolean;
  /**
   * 多选框的值
   */
  value?: string | number;
  /**
   * 值变化时触发
   */
  onChange?: (checked: boolean, ev: React.ChangeEvent<HTMLInputElement>) => void;
}

export interface CheckboxGroupProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
  /**
   * 多选框组选中固定值（受控）
   */
  selected?: Array<string | number> | string | number;
  /**
   * 多选框组选中默认值（非受控）
   * @default 第一项
   */
  defaultSelected?: Array<string | number> | string | number;
  /**
   * 按钮多选框全局尺寸
   * @default medium
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * 多选框组是否全局禁用
   */
  disabled?: boolean;
  /**
   * 选中某一项时触发
   */
  onChange?: (
    value: Array<string | number> | string | number,
    ev: React.ChangeEvent<HTMLInputElement>,
  ) => void;
}

export interface CheckboxContextValue {
  inject: (props: CheckboxProps) => CheckboxProps;
}
