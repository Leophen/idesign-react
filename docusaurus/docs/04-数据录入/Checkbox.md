---
title: Checkbox 多选框
---

在一组可选项中进行多项选择时；

单独使用可以表示两种状态之间的切换，和 `switch` 类似。区别在于切换 `switch` 会直接触发状态改变，而 `checkbox` 一般用于状态标记，需要和提交操作配合。

## 引入

```js
import {Checkbox} from 'idesign';
```

## 基本用法

```jsx live fff
function App() {
    const [checked, setChecked] = useState(true);

    const handleChange = (e) => {
        console.log(e);
        setChecked(e.target.checked);
    };

    return (
        <div className="demo-column">
            <Checkbox>无默认值</Checkbox>
            <Checkbox defaultChecked={checked}>有默认值（非受控）</Checkbox>
            <Checkbox checked={checked}>有固定值（受控）</Checkbox>
            <Checkbox checked={checked} onChange={handleChange}>
                通用方法
            </Checkbox>
        </div>
    );
}
```

## 禁用状态

```jsx live fff
<div className="demo-column">
    <Checkbox defaultChecked={false} disabled>disabled</Checkbox>
    <Checkbox defaultChecked disabled>disabled</Checkbox>
</div>
```

## 多选框组

```jsx live fff
function App() {
    function onChange(checkedValues) {
        console.log('checked = ', checkedValues);
    }

    const plainOptions = ['Apple', 'Pear', 'Orange'];

    const options = [
        {label: 'Apple', value: 'Apple'},
        {label: 'Pear', value: 'Pear'},
        {label: 'Orange', value: 'Orange'}
    ];

    const optionsWithDisabled = [
        {label: 'Apple', value: 'Apple'},
        {label: 'Pear', value: 'Pear'},
        {label: 'Orange', value: 'Orange', disabled: false}
    ];

    return (
        <div className="demo-column">
            <Checkbox.Group options={plainOptions} defaultValue={['Apple']} onChange={onChange} />
            <Checkbox.Group options={options} defaultValue={['Pear']} onChange={onChange} />
            <Checkbox.Group
                options={optionsWithDisabled}
                disabled
                defaultValue={['Apple']}
                onChange={onChange}
            />
        </div>
    );
}
```

## 控制全选

```jsx live fff
function App() {
    const plainOptions = ['Apple', 'Pear', 'Orange'];
    const defaultCheckedList = ['Apple', 'Orange'];

    const [checkedList, setCheckedList] = useState(defaultCheckedList);
    const [indeterminate, setIndeterminate] = useState(true);
    const [checkAll, setCheckAll] = useState(false);

    const onChange = list => {
        setCheckedList(list);
        setIndeterminate(!!list.length && list.length < plainOptions.length);
        setCheckAll(list.length === plainOptions.length);
    };

    const onCheckAllChange = e => {
        setCheckedList(e.target.checked ? plainOptions : []);
        setIndeterminate(false);
        setCheckAll(e.target.checked);
    };

    return (
        <div className="demo-column">
            <Checkbox
                indeterminate={indeterminate}
                onChange={onCheckAllChange}
                checked={checkAll}
            >
                Check all
            </Checkbox>
            <Checkbox.Group options={plainOptions} value={checkedList} onChange={onChange} />
        </div>
    );
}
```

## Checkbox Button

```jsx live fff
function App() {
    function onChange(e) {
        console.log(`checked = ${e.target.checked}`);
    }

    return (
        <div className="demo-row">
            <Checkbox.Button onChange={onChange}>Checkbox</Checkbox.Button>
            <Checkbox.Button onChange={onChange}>我的数据很长很长啊啊啊啊</Checkbox.Button>
            <Checkbox.Button disabled onChange={onChange}>disabled</Checkbox.Button>
            <Checkbox.Button disabled checked onChange={onChange}>disabled</Checkbox.Button>
        </div>
    );
}
```

## Checkbox Button 组

方便的从数组生成 Checkbox 组。

```jsx live fff
function App() {
    function onChange(checkedValues) {
        console.log('checked = ', checkedValues);
    }

    const plainOptions = ['Apple', 'Pear', 'Orange'];
    const options = [
        {label: 'Apple', value: 'Apple'},
        {label: 'Pear', value: 'Pear'},
        {label: 'Orange', value: 'Orange'}
    ];
    const optionsWithDisabled = [
        {label: 'Apple', value: 'Apple'},
        {label: 'Pear', value: 'Pear'},
        {label: 'Orange', value: 'Orange', disabled: false}
    ];
    const optionsWithDisabled2 = [
        {label: 'data1', value: 'data1'},
        {label: 'data2', value: 'data2', disabled: true},
        {label: 'data3', value: 'data3'}
    ];

    const optionsBigData = [];
    for (let i = 1; i < 20; i++) {
        optionsBigData.push({
            label: `data${i}`,
            value: `data${i}`
        });
    }

    return (
        <div className="demo-column">
            <Checkbox.Group optionType="button" options={plainOptions} defaultValue={['Apple']} onChange={onChange} />
            <Checkbox.Group optionType="button" options={options} defaultValue={['Pear']} onChange={onChange} />
            <Checkbox.Group
                optionType="button"
                options={optionsWithDisabled}
                disabled
                defaultValue={['Pear']}
                onChange={onChange}
            />
            <Checkbox.Group
                optionType="button"
                options={optionsWithDisabled2}
                defaultValue={['data1']}
                onChange={onChange}
            />
            <Checkbox.Group
                optionType="button"
                style={{flexWrap: 'wrap'}}
                options={optionsBigData}
                defaultValue={['data1']}
                onChange={onChange}
            />
        </div>
    );
}
```

## API

### Checkbox

| 参数           | 说明                                    | 类型                | 默认值  |
| -------------- | --------------------------------------- | ------------------- | ------- |
| autoFocus      | 自动获取焦点                            | `boolean`           | `false` |
| checked        | 指定当前是否选中                        | `boolean`           | `false` |
| defaultChecked | 初始是否选中                            | `boolean`           | `false` |
| disabled       | 失效状态                                | `boolean`           | `false` |
| indeterminate  | 设置 indeterminate 状态，只负责样式控制 | `boolean`           | `false` |
| onChange       | 变化时回调函数                          | `function(e:Event)` | `--`    |

### Checkbox Group

| 参数         | 说明                                                         | 类型                     | 默认值    |
| ------------ | ------------------------------------------------------------ | ------------------------ | --------- |
| defaultValue | 默认选中的选项                                               | `string[]`               | `[]`      |
| disabled     | 整组失效                                                     | `boolean`                | `false`   |
| name         | CheckboxGroup 下所有 `input[type="checkbox"]` 的 `name` 属性 | `string`                 | `--`      |
| options      | 指定可选项                                                   | `string[]〡Option[]`     | `[]`      |
| value        | 指定选中的选项                                               | `string[]`               | `[]`      |
| onChange     | 变化时回调函数                                               | `function(checkedValue)` | `--`      |
| optionType   | 用于设置 Checkbox options 类型                               | `default〡button`        | `default` |

### Checkbox Option

```typescript
interface Option {
    label: string;
    value: string;
    disabled?: boolean;
}
```

### Checkbox 方法

| 名称    | 描述     |
| ------- | -------- |
| blur()  | 移除焦点 |
| focus() | 获取焦点 |
