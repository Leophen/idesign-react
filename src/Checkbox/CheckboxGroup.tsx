import React, { useState, useEffect } from 'react';
import classNames from 'classnames';
import './index.scss';
import { CheckboxContextValue, CheckboxGroupProps, CheckboxProps } from './type';

export const CheckboxContext = React.createContext<CheckboxContextValue>(null as any);

// 多选框组
const CheckboxGroup: React.FC<CheckboxGroupProps> = (props) => {
  const { children = '', className, style, currentValue, size, disabled, onChange } = props;

  const [groupValue, setGroupValue] = useState(currentValue);
  let groupCheckedArr: Set<any> = new Set([].concat(groupValue as any));

  useEffect(() => {
    setGroupValue(currentValue);
    groupCheckedArr = new Set([].concat(groupValue as any));
  }, [currentValue]);

  // 注入每一项多选框的 context
  const context: CheckboxContextValue = {
    // 将多选框组的 props 注入单项多选框的方法
    inject: (singleCheckboxProps: CheckboxProps) => {
      // 拿到单项多选框的 value，方便与多选框组的 groupValue 做比较
      const singleCheckboxValue = singleCheckboxProps.value;
      return {
        ...singleCheckboxProps,
        size,
        disabled,
        checked: groupCheckedArr.has(singleCheckboxValue),
        onChange(checked, e) {
          e.persist();
          // 触发单项多选框传入的 onChange
          if (typeof singleCheckboxProps.onChange === 'function') {
            singleCheckboxProps.onChange(checked, e);
          }
          // 设置多选框组的 groupCheckedArr
          if (checked) {
            groupCheckedArr.add(singleCheckboxValue);
          } else {
            groupCheckedArr.delete(singleCheckboxValue);
          }
          // 触发多选框组传入的 onChange
          onChange?.(Array.from(groupCheckedArr) as any, e);
        },
      };
    },
  };

  return (
    <CheckboxContext.Provider value={context}>
      <div className={classNames('i-checkbox-group', className)} style={{ ...style }}>
        {children}
      </div>
    </CheckboxContext.Provider>
  );
};

export default CheckboxGroup;
