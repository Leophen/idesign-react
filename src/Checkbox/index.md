---
nav:
  title: 组件
  path: /components
group:
  title: 表单组件
  order: 3
order: 3
---

# Checkbox 多选框

多选框是一个选择控件，允许用户通过单击在选中和未选中之间切换。

## 基本用法

最简单的多选框形式。

```tsx
import React, { useState } from 'react';
import { Checkbox } from 'idesign-react';

const App = () => {
  const [checked, setChecked] = useState(false);

  const handleChange = (val, e) => {
    console.log(val, e);
    setChecked(val);
  };

  return (
    <div className="idesign-demo-block-row">
      <Checkbox checked={true}>已选中</Checkbox>
      <Checkbox checked={checked} onChange={handleChange}>
        {checked ? '已选中' : '未选中'}
      </Checkbox>
    </div>
  );
};

export default App;
```

## 多选框组

使用 `Checkbox.Group` 可以包裹 `Checkbox` 形成一组多选框组。

```tsx
import React, { useState } from 'react';
import { Checkbox } from 'idesign-react';

const App = () => {
  const [currentValue, setCurrentValue] = useState(['gz']);

  const handleChange = (val, e) => {
    console.log(val, e);
    setCurrentValue(val);
  };

  return (
    <Checkbox.Group currentValue={currentValue} onChange={handleChange}>
      <Checkbox value="bj">北京</Checkbox>
      <Checkbox value="sh">上海</Checkbox>
      <Checkbox value="gz">广州</Checkbox>
      <Checkbox value="sz">深圳</Checkbox>
    </Checkbox.Group>
  );
};

export default App;
```

## 不同尺寸

`Checkbox` 多选框提供 `size` 属性控制尺寸，`Checkbox.Group` 也有 `size` 属性可以全局控制每一项多选框的尺寸。

```tsx
import React, { useState } from 'react';
import { Checkbox } from 'idesign-react';

const App = () => {
  const [currentValue, setCurrentValue] = useState(['gz']);

  const handleChange = (val) => {
    setCurrentValue(val);
  };

  return (
    <div className="idesign-demo-block-column">
      <div className="idesign-demo-block-row">
        <Checkbox size="small">small</Checkbox>
        <Checkbox size="medium">medium</Checkbox>
        <Checkbox size="large">large</Checkbox>
      </div>

      <div className="idesign-demo-block-row">
        <Checkbox.Group size="small" currentValue={currentValue} onChange={handleChange}>
          <Checkbox value="bj">北京</Checkbox>
          <Checkbox value="sh">上海</Checkbox>
          <Checkbox value="gz">广州</Checkbox>
          <Checkbox value="sz">深圳</Checkbox>
        </Checkbox.Group>
      </div>

      <div className="idesign-demo-block-row">
        <Checkbox.Group size="medium" currentValue={currentValue} onChange={handleChange}>
          <Checkbox value="bj">北京</Checkbox>
          <Checkbox value="sh">上海</Checkbox>
          <Checkbox value="gz">广州</Checkbox>
          <Checkbox value="sz">深圳</Checkbox>
        </Checkbox.Group>
      </div>

      <div className="idesign-demo-block-row">
        <Checkbox.Group size="large" currentValue={currentValue} onChange={handleChange}>
          <Checkbox value="bj">北京</Checkbox>
          <Checkbox value="sh">上海</Checkbox>
          <Checkbox value="gz">广州</Checkbox>
          <Checkbox value="sz">深圳</Checkbox>
        </Checkbox.Group>
      </div>
    </div>
  );
};

export default App;
```

## 禁用状态

`Checkbox` 提供 `disabled` 属性控制多选框为禁用状态，`Checkbox.Group` 可以全局控制每一项为禁用状态。

```tsx
import React, { useState } from 'react';
import { Checkbox } from 'idesign-react';

const App = () => {
  const [currentValue, setCurrentValue] = useState(['gz']);

  const handleChange = (val) => {
    setCurrentValue(val);
  };

  return (
    <div className="idesign-demo-block-column">
      <h3>单项禁用</h3>

      <div className="idesign-demo-block-row">
        <Checkbox disabled>未选中</Checkbox>
      </div>

      <h3>多选框组局部禁用</h3>

      <div className="idesign-demo-block-row">
        <Checkbox.Group currentValue={currentValue} onChange={handleChange}>
          <Checkbox value="bj">北京</Checkbox>
          <Checkbox value="sh">上海</Checkbox>
          <Checkbox value="gz">广州</Checkbox>
          <Checkbox value="sz" disabled>
            深圳
          </Checkbox>
        </Checkbox.Group>
      </div>

      <h3>多选框组全局禁用</h3>

      <div className="idesign-demo-block-row">
        <Checkbox.Group disabled currentValue={currentValue} onChange={handleChange}>
          <Checkbox value="bj">北京</Checkbox>
          <Checkbox value="sh">上海</Checkbox>
          <Checkbox value="gz">广州</Checkbox>
          <Checkbox value="sz">深圳</Checkbox>
        </Checkbox.Group>
      </div>
    </div>
  );
};

export default App;
```

## Checkbox API

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| children | 按钮内容 | `ReactNode` | `--` |
| className | 类名 | `string` | `--` |
| style | 自定义样式 | `CSSProperties` | `--` |
| size | 多选框尺寸 | `"small"〡"medium"〡"large"` | `"medium"` |
| value | 多选框的值 | `string〡number` | `--` |
| checked | 是否选中 | `boolean` | `false` |
| disabled | 是否禁用组件 | `boolean` | `false` |
| onChange | 值变化时触发 | `(checked: boolean, context: React.ChangeEvent<HTMLInputElement>) => void` | `--` |

## CheckboxGroup API

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| children | 按钮内容 | `ReactNode` | `--` |
| className | 类名 | `string` | `--` |
| style | 自定义样式 | `CSSProperties` | `--` |
| currentValue | 多选框组选中的值 | `string〡number` | `--` |
| size | 多选框全局尺寸 | `"small"〡"medium"〡"large"` | `"medium"` |
| disabled | 多选框组是否全局禁用 | `boolean` | `false` |
| onChange | 选中某一项时触发 | `(value: Array<string〡number>〡string〡number, context: React.ChangeEvent<HTMLInputElement>) => void` | `--` |
