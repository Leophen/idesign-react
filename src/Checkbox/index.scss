.i-checkbox {
  position: relative;
  display: inline-flex;
  align-items: center;
  color: var(--i-font-2);
  cursor: pointer;
  &.i-checkbox--size-small {
    .i-checkbox__input {
      transform: scale(0.8);
      transform-origin: left;
    }
    .i-checkbox__label {
      margin: 0 4px;
    }
  }
  &.i-checkbox--size-large {
    .i-checkbox__input {
      transform: scale(1.2);
      transform-origin: left;
    }
    .i-checkbox__label {
      margin: 0 12px;
    }
  }
  &:hover {
    .i-checkbox__input {
      border-color: var(--i-primary-hover);
    }
  }
  &.i-checkbox-is-checked {
    margin-left: inherit;
    .i-checkbox__input {
      background-color: var(--i-primary);
      border-color: var(--i-primary);
      transition: background-color 0.1s cubic-bezier(0.82, 0, 1, 0.9);
      &::after {
        opacity: 1;
      }
    }
  }
  &.i-checkbox-is-disabled {
    cursor: not-allowed;
    &:hover,
    &.i-checkbox-is-checked {
      .i-checkbox__input {
        border-color: var(--i-border);
      }
    }
    .i-checkbox__input {
      background-color: var(--i-bg-disabled);
      &::after {
        border-color: var(--i-border);
      }
    }
    .i-checkbox__label {
      color: var(--i-font-disabled);
    }
  }
}

.i-checkbox__former {
  position: absolute;
  box-sizing: border-box;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  overflow: hidden;
  border: 0;
  outline: 0;
  appearance: none;
  clip: rect(0 0 0 0);
}

.i-checkbox__input {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  width: 16px;
  height: 16px;
  vertical-align: middle;
  background-color: var(--i-bg);
  border: 1px solid var(--i-border);
  border-radius: 4px;
  transition: all 0.2s ease-out;
  &::after {
    position: absolute;
    top: 7px;
    left: 3px;
    width: 4px;
    height: 8px;
    background: transparent;
    border: 2px solid #fff;
    border-top: 0;
    border-left: 0;
    border-radius: 0 0 1px;
    transform: rotate(45deg) scale(1) translate(-60%, -50%);
    opacity: 0;
    content: '';
  }
}

.i-checkbox__label {
  display: inline-block;
  margin: 0 8px;
  font-size: 14px;
  vertical-align: middle;
  user-select: none;
}

.i-checkbox-group {
  .i-checkbox {
    margin-right: 16px;
  }
}
