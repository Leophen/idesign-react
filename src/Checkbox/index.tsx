import React, { useState, useContext, useEffect } from 'react';
import classNames from 'classnames';
import './index.scss';
import CheckboxGroup, { CheckboxContext } from './CheckboxGroup';
import { CheckboxProps } from './type';

// 单一多选框
const Checkbox: React.FC<CheckboxProps> & { Group: React.ElementType } = (props) => {
  // 存在多选框组时从 Context 注入 checked 覆盖原 checked
  const context = useContext(CheckboxContext);
  const newProps = context ? context.inject(props) : props;

  const { children = '', className, style, checked = false, value, onChange, ...restProps } = newProps;

  const size = newProps.size || props.size;
  const disabled = newProps.disabled || props.disabled || false;

  const [innerChecked, setInnerChecked] = useState(checked);

  useEffect(() => {
    setInnerChecked(checked);
  }, [checked]);

  const handleCheck = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (disabled) return;
    const checkedValue = e.currentTarget.checked;
    // 多选框组时使用 context 提供的修改 checked 方法同步整组的 currentValue
    if (typeof newProps.onChange === 'function') {
      newProps.onChange(checkedValue, e);
    }
  };

  return (
    <label
      className={classNames(
        'i-checkbox',
        innerChecked && 'i-checkbox-is-checked',
        disabled && 'i-checkbox-is-disabled',
        size && `i-checkbox--size-${size}`,
        className,
      )}
      style={{ ...style }}
      {...restProps}
    >
      <input
        readOnly
        type="checkbox"
        className={'i-checkbox__former'}
        checked={innerChecked}
        disabled={disabled}
        value={value}
        onClick={(e) => e.stopPropagation()}
        onChange={(e) => handleCheck(e)}
      />
      <span className={'i-checkbox__input'} />
      <span className={'i-checkbox__label'}>{children}</span>
    </label>
  );
};

Checkbox.Group = CheckboxGroup;

Checkbox.displayName = 'Checkbox';
CheckboxGroup.displayName = 'CheckboxGroup';

export default Checkbox;
