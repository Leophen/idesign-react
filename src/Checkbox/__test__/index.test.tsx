import { render, fireEvent } from '@testing-library/react';
import Checkbox from '../index';

describe('Checkbox 组件测试', () => {
  it('create', () => {
    const { asFragment } = render(<Checkbox />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('checked & children', () => {
    const { container, queryByText } = render(<Checkbox checked={true}>广州</Checkbox>);
    expect(container.firstChild).toHaveClass('i-checkbox', 'i-checkbox-is-checked');
    expect(queryByText('广州')).toBeInTheDocument();
  });

  it('defaultChecked', () => {
    const { container } = render(<Checkbox defaultChecked={true} />);
    expect(container.firstChild).toHaveClass('i-checkbox', 'i-checkbox-is-checked');
  });

  it('disabled', () => {
    const fn = jest.fn();
    const { container } = render(<Checkbox disabled={true} onChange={fn} />);
    expect(container.firstChild).toHaveClass('i-checkbox-is-disabled');
    fireEvent.click(container.firstChild);
    expect(fn).toBeCalledTimes(0);
  });

  it('size', () => {
    const { container } = render(<Checkbox size="small" />);
    expect(container.firstChild).toHaveClass('i-checkbox--size-small');
  });
});

describe('CheckboxGroup 组件测试', () => {
  it('selected', () => {
    const { container } = render(
      <Checkbox.Group selected={['sz']}>
        <Checkbox value="gz">广州</Checkbox>
        <Checkbox value="sz">深圳</Checkbox>
      </Checkbox.Group>,
    );
    expect(container.firstChild.lastChild).toHaveClass('i-checkbox-is-checked');
  });

  it('defaultSelected', () => {
    const { container } = render(
      <Checkbox.Group defaultSelected={['gz']}>
        <Checkbox value="gz">广州</Checkbox>
        <Checkbox value="sz">深圳</Checkbox>
      </Checkbox.Group>,
    );
    expect(container.firstChild.firstChild).toHaveClass('i-checkbox-is-checked');
  });

  it('disabled', () => {
    const fn = jest.fn();
    const { container } = render(
      <Checkbox.Group disabled>
        <Checkbox value="gz">广州</Checkbox>
        <Checkbox value="sz">深圳</Checkbox>
      </Checkbox.Group>,
    );
    expect(container.firstChild.firstChild).toHaveClass('i-checkbox-is-disabled');
    fireEvent.click(container.firstChild.firstChild);
    expect(fn).toBeCalledTimes(0);
  });

  it('size', () => {
    const { container } = render(
      <Checkbox.Group size="small">
        <Checkbox value="gz">广州</Checkbox>
        <Checkbox value="sz">深圳</Checkbox>
      </Checkbox.Group>,
    );
    expect(container.firstChild.firstChild).toHaveClass('i-checkbox--size-small');
  });
});

