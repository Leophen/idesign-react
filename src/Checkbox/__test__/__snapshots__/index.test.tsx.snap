// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox 组件测试 create 1`] = `
<DocumentFragment>
  <label
    class="i-checkbox"
  >
    <input
      class="i-checkbox__former"
      readonly=""
      type="checkbox"
      value=""
    />
    <span
      class="i-checkbox__input"
    />
    <span
      class="i-checkbox__label"
    />
  </label>
</DocumentFragment>
`;
